data/modules/Scoop/Scoop.lua

```diff
@@ -0,0 +1,837 @@
+		if #debris > 0 then
+			spawnDebris(debris, flavour.amount, planet, radius * 1.2, radius * 3.5, mission_time)
+		end
+	end
+end
+
+local onLeaveSystem = function (ship)
+	if ship:IsPlayer() then
+		for ref, mission in pairs(missions) do
+			mission.destination = nil
+			mission.police = nil
+			if mission.pirate then
+				mission.pirate = nil
+				mission.status = "FAILED"
+			end
+			for i, e in pairs(mission.debris) do
+				e.body = nil
+			end
+		end
+		planets = nil
+		flavours[LEGAL].cargo_type = nil
+		flavours[ILLEGAL].cargo_type = nil
+	end
+end
+
+local onReputationChanged = function (oldRep, oldKills, newRep, newKills)
+	for ref, ad in pairs(ads) do
+		local oldQualified = isQualifiedFor(oldRep, ad)
+		if isQualifiedFor(newRep, ad) ~= oldQualified then
+			Event.Queue("onAdvertChanged", ad.station, ref);
+		end
+	end
+end
+
+local buildMissionDescription = function(mission)
+	local ui = require 'pigui'
+	local desc = {}
+
+	desc.description = mission.introtext:interp({
+		client = mission.client.name,
+		shipid = mission.pirate_label,
+		star   = mission.star:GetSystemBody().name,
+		planet = mission.planet:GetSystemBody().name,
+		cash   = Format.Money(math.abs(mission.reward), false)
+	})
+
+	desc.details = {
+		{ l.CLIENT, mission.client.name },
+		{ l.SPACEPORT, mission.station:GetSystemBody().name },
+		mission.pirate and { l.SHIP, mission.pirate.label } or false,
+		false,
+		{ l.DEADLINE, ui.Format.Date(mission.due) }
+	}
+
+	desc.client = mission.client
+	desc.location = mission.destination or nil
+	if mission.deliver_to_ship then
+		desc.returnLocation = mission.pirate or mission.station
+	end
+
+	return desc
+end
+
+local loaded_data
+
+local onGameStart = function ()
+	ads = {}
+	missions = {}
+
+	if loaded_data and loaded_data.ads then
+
+		for k, ad in pairs(loaded_data.ads) do
+			local ref = ad.station:AddAdvert({
+				description = ad.desc,
+				icon        = ad.id == "RESCUE" and "searchrescue" or "haul",
+				onChat      = onChat,
+				onDelete    = onDelete,
+				isEnabled   = isEnabled
+			})
+			ads[ref] = ad
+		end
+
+		missions = loaded_data.missions
+
+		loaded_data = nil
+
+		for ref, mission in pairs(missions) do
+			if mission.deliver_to_ship then
+				mission.in_progress = true
+				transferCargo(mission)
+			end
+		end
+	end
+end
+
+local onGameEnd = function ()
+	planets = nil
+	flavours[LEGAL].cargo_type = nil
+	flavours[ILLEGAL].cargo_type = nil
+end
+
+local serialize = function ()
+	return { ads = ads, missions = missions }
+end
+
+local unserialize = function (data)
+	loaded_data = data
+end
+
+Event.Register("onCreateBB", onCreateBB)
+Event.Register("onUpdateBB", onUpdateBB)
+Event.Register("onShipEquipmentChanged", onShipEquipmentChanged)
+Event.Register("onShipDocked", onShipDocked)
+Event.Register("onShipUndocked", onShipUndocked)
+Event.Register("onShipHit", onShipHit)
+Event.Register("onShipDestroyed", onShipDestroyed)
+Event.Register("onCargoDestroyed", onCargoDestroyed)
```

Comment:
I'd recommend adding a function that reacts to the mission cargo being jettisoned from the player's hold with `Event.Register("onJettisoned")`. I leave it at your discretion whether that is grounds for failing the mission or simply a strongly-worded CommsMessage encouraging the player to scoop it back up.